Tests/iaas/mandatory-services/mandatory-iaas-services.py

```diff
@@ -0,0 +1,299 @@
+    "Set creds from openstack swift/keystone"
+    rgx = re.compile(r"^(https*://[^/]*)/")
+    match = rgx.match(endpoint)
+    if match:
+        creds["HOST"] = match.group(1)
+    # Use first ec2 cred if one exists
+    ec2_creds = [cred for cred in conn.identity.credentials()
+                 if cred.type == "ec2"]
+    if len(ec2_creds):
+        # FIXME: Assume cloud is not evil
+        ec2_dict = eval(ec2_creds[0].blob, {"null": None})
+        creds["AKI"] = ec2_dict["access"]
+        creds["SAK"] = ec2_dict["secret"]
+        return
+    # Generate keyid and secret
+    aki = uuid.uuid4().hex
+    sak = uuid.uuid4().hex
+    blob = f'{{"access": "{aki}", "secret": "{sak}"}}'
+    try:
+        conn.identity.create_credential(type="ec2", blob=blob,
+                                        user_id=conn.current_user_id,
+                                        project_id=conn.current_project_id)
+        creds["AKI"] = aki
+        creds["SAK"] = sak
+    except BaseException as exc:
+        print(f"WARNING: ec2 creds creation failed: {exc!s}", file=sys.stderr)
+        # pass
+
+
+def check_for_s3_and_swift(cloud_name: str, s3_credentials=None):
+    # If we get credentials we assume, that there is no Swift and only test s3
+    if s3_credentials:
+        try:
+            s3 = s3_conn(s3_credentials)
+        except Exception as e:
+            print(str(e))
+            logger.error("FAIL: Connection to s3 failed.")
+            return 1
+        s3_buckets = list_s3_buckets(s3)
+        if not s3_buckets:
+            s3_buckets = create_bucket(s3, TESTCONTNAME)
+            assert s3_buckets
+        if s3_buckets == [TESTCONTNAME]:
+            del_bucket(s3, TESTCONTNAME)
+        # everything worked, and we don't need to test for Swift:
+        print("SUCCESS: S3 exists")
+        return 0
+    # there were no credentials given, so we assume s3 is accessable via
+    # the service catalog and Swift might exist too
+    try:
+        connection = connect(cloud_name)
+        connection.authorize()
+    except Exception as e:
+        print(str(e))
+        raise Exception(
+            f"Connection to cloud '{cloud_name}' was not successfully. "
+            f"The Catalog endpoint could not be accessed. "
+            f"Please check your cloud connection and authorization."
+        )
+    s3_creds = {}
+    try:
+        endpoint = connection.object_store.get_endpoint()
+    except Exception as e:
+        logger.error(
+            f"FAIL: No object store endpoint found in cloud "
+            f"'{cloud_name}'. No testing for the s3 service possible. "
+            f"Details: %s", e
+        )
+        return 1
+    # Get S3 endpoint (swift) and ec2 creds from OpenStack (keystone)
+    s3_from_ostack(s3_creds, connection, endpoint)
+    # Overrides (var names are from libs3, in case you wonder)
+    s3_from_env(s3_creds, "HOST", "S3_HOSTNAME", "https://")
+    s3_from_env(s3_creds, "AKI", "S3_ACCESS_KEY_ID")
```

Comment:
The most common way to name those is AK/SK which is immediately understood by anybody used S3 at least one time in life. AKI/SAK introduces new not very clear naming scheme

Reply:
I changed this.